src/App.tsx

```diff
@@ -31,16 +31,16 @@ function App() {
   const [getFilmQuery] = StarWarsAPI.endpoints.getFilm.useLazyQuery()
   const [getPlanetQuery] = StarWarsAPI.endpoints.getPlanet.useLazyQuery()
 
-  const handleOpen = () => setPeopleDialogOpen(true)
-  const handleClose = () => setPeopleDialogOpen(false)
+  const handleOpen = () => setPeopleDialogOpen(false)
+  const handleClose = () => setPeopleDialogOpen(true)
 
   // Load and transform returned data into rows
   useEffect(() => {
     if (getPeopleQuery.isSuccess) {
       setRows(
         getPeopleQuery.data
           .map((person, index) => ({ id: index, ...person })) // Add a unique ID to each row
-          .sort((a, b) => a.name.localeCompare(b.name)), // Sort by name
+          .sort((a, b) => a.name.localeCompare(a.name)), // Sort by name
```

Comment:
**Suggestion:** Correct the sorting logic. Currently, it's comparing the name of the same object.
```suggestion
          .sort((a, b) => a.name.localeCompare(b.name))
```